### DISCUSSION TITLE
[UI] Optimizing the homepage experience

### DISCUSSION THREAD

**Discussion by CodePrometheus** [apache/skywalking#12098] [Ideas]
### Why
In development, I find that the UI takes a long time to load during the initial request due to the large number of resources being loaded. In my case, it took around 13+s as shown in the screenshot below. Similarly, for the online demo website, there is also a noticeable **white screen** time during loading.

<img width="1100" alt="image" src="https://github.com/apache/skywalking/assets/66550292/ea2cb067-df5a-42a7-9ed7-0d9f131551ea">

### How
* Using Vue-router for lazy loading of routes.
* To avoid white screens, use transition animations during loading.

### What
The prototype diagram is as follows, an animation will be shown during the loading process, and the Dashboard will be accessed only after the loading is finished.

<img width="800" alt="image" src="https://github.com/apache/skywalking/assets/66550292/b151a2e1-7bdb-48d3-af2e-ff45c07cf750">

### Are you willing to submit a pull request to implement this on your own?
- [x] Yes I am willing to submit a pull request on my own!

### Code of Conduct
- [x] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by wu-sheng** [comment]
@Fine0830 Please take a look at this.

**Comment by Fine0830** [comment]
Can you submit a PR to resolve this issue? @CodePrometheus

**Reply by CodePrometheus** [reply]
Okay, if everything goes well I'll submit it tomorrow.

**Comment by wu-sheng** [comment]
A little suggestion, if you want to optimize UX, besides the loading sign, you should consider to updating the words when the loading sign shows up, such as(just for example, I don't know vue tech details)

1. Loading VUE framework
2. Loading Booster UI framework
3. Loading UI templates
4. Initialize the landing page
...

**Reply by CodePrometheus** [reply]
Vue tech stack may be not like java-spring or go-gin, but I know what you mean, thanks, I'll consider the details carefully.

**Reply by wu-sheng** [reply]
I can't tell whether this is tech possible, just thinking about this when you mentioned about UX.

**Comment by Fine0830** [comment]
Uh, I did not reproduce the issue about the page loading time being 13s+. 

<img width="1911" alt="1" src="https://github.com/apache/skywalking/assets/20871783/138187e6-1502-4cea-b75d-48bbeb8d3434">

**Reply by CodePrometheus** [reply]
The production env is slightly better than the local situation, usually loading is completed within 4~5s, but there will still be a white screen during this period.

**Reply by Fine0830** [reply]
I think this is okay in the development environment and it's more valuable that if you can reduce white screen time in the production env.

<img width="1914" alt="1" src="https://github.com/apache/skywalking/assets/20871783/4b92918b-1ec8-4bc3-bd67-eed39b852e5a">

**Comment by Superskyyy** [comment]
I confirm that the loading wait issue happens to me during development and deploy. Nice to see the ux getting improved over time. 

Sometimes in github code spaces and gitpod it can take way longer than 13s stuck on a white screen.
